controller/src/main/java/io/pravega/controller/server/eventProcessor/requesthandlers/DeleteReaderGroupTask.java

```diff
@@ -0,0 +1,79 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.controller.server.eventProcessor.requesthandlers;
+
+import com.google.common.base.Preconditions;
+
+import io.pravega.client.stream.ReaderGroupConfig;
+import io.pravega.client.stream.Stream;
+import io.pravega.common.concurrent.Futures;
+import io.pravega.controller.store.stream.RGOperationContext;
+import io.pravega.controller.store.stream.StreamMetadataStore;
+import io.pravega.controller.task.Stream.StreamMetadataTasks;
+import io.pravega.shared.NameUtils;
+import io.pravega.shared.controller.event.DeleteReaderGroupEvent;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.Iterator;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ScheduledExecutorService;
+
+/**
+ * Request handler for executing a delete operation for a ReaderGroup.
+ */
+@Slf4j
+public class DeleteReaderGroupTask implements ReaderGroupTask<DeleteReaderGroupEvent> {
+
+    private final StreamMetadataStore streamMetadataStore;
+    private final StreamMetadataTasks streamMetadataTasks;
+    private final ScheduledExecutorService executor;
+
+    public DeleteReaderGroupTask(final StreamMetadataTasks streamMetaTasks,
+                                 final StreamMetadataStore streamMetaStore,
+                                 final ScheduledExecutorService executor) {
+        Preconditions.checkNotNull(streamMetaStore);
+        Preconditions.checkNotNull(streamMetaTasks);
+        Preconditions.checkNotNull(executor);
+        this.streamMetadataStore = streamMetaStore;
+        this.streamMetadataTasks = streamMetaTasks;
+        this.executor = executor;
+    }
+
+    @Override
+    public CompletableFuture<Void> execute(final DeleteReaderGroupEvent request) {
+      String scope = request.getScope();
```

Comment:
do remember, our event processor has at least once processing guarantees. so the events can be replayed. 
which means the delete rg can be replayed. 
we do want to make its processing idempotent and not perform an unwanted delete in a subsequent replay of the event. 

so this delete `event` should include the "generation" or starting segmentsegmentnumber of the RG to be deleted. so that when it is processed we only delete the intended RG.. if it was already deleted previously and the current RG has a higher generation number then we ignore this event and do nothing.

Reply:
fixed